### PR TITLE
(Feat): Finetuning

### PR DESCRIPTION
**Pull request by SimWalther**
This branch add finetuning.

There's still some issues:
- Is it the right way to train a baseline model if there's isn't any model or should we assume that there's one?
- Should the metrics of the baseline model be loaded from a previous run metric file (but we need to compute the baseline model performances on the new set in any case)
- Is it correct to save the finetuned model as the new baseline model automaticaly if the model is better or should there be some manual selection of new baseline models?

### CODE REVIEW

**Review by Fastium — COMMENTED** [review]
@SimWalther Du coup, il n'y a pas de condition pour fine-tune ou entrainer à nouveau complètement le modèle.

Je me demande si ce n'est pas mieux de faire une étape fine-tune et une étape train. Sinon, on peut mettre un flag qui dicte ce qu'on veut faire. Parce que si j'ai bien compris, pour l'instant le model n'est que fine-tune.

**Comment by SimWalther** [comment]
@Fastium Mon problème était justement de trouver comment faire ça bien, actuellement on train "from scratch" si il n'y aucun modèle défini.

Faire des étapes séparées ça pourrait être bien et on pourrait donc mettre le modèle baseline comme un output de l'étape en question. Je voulais juste pas qu'on doive réentraîner un modèle baseline à chaque fois.

**Comment by Fastium** [comment]
@SimWalther , alors je penses qu on peut pudh cette branche dans main, puisque t as fait pas mal de changement. 
Et pour le ré-entrainement, on peut le trigger soit en mesurant les perf, soit avec le temps. Qu'est-ce que tu en dit ?

**Comment by SimWalther** [comment]
@Fastium Je viens de mettre en place ces changements. Si quelqu'un peut review on peut merge ça.

**Review by Fastium — APPROVED** [review]

**Comment by Fastium** [comment]
Tu peux merge dans main quand tu veux
